--- JaroMail-e3c34f9020ba5ba2da783ce56849a1ccd0a11e8d/src/zlibs/filters
+++ JaroMail-e3c34f9020ba5ba2da783ce56849a1ccd0a11e8d/src/zlibs/filters
@@ -62,9 +62,6 @@
     ${=mkdir} "$MAILDIRS/Accounts"
     cp "$WORKDIR"/Accounts/* "$MAILDIRS/Accounts/" }
 
-    { test -r "$MAILDIRS"/Manual.pdf } || {
-    cp "$WORKDIR"/jaromail-manual.pdf "$MAILDIRS"/Manual.pdf }
-
     { test -r "$MAILDIRS"/Identity.txt } || {
     cp "$WORKDIR"/Identity.txt "$MAILDIRS"/Identity.txt }
 
