diff -dPNur CodeAnalyst-Linux-2.8.29.orig/src/ca/gui/oprofile_interface.cpp CodeAnalyst-Linux-2.8.29.orig.detection/src/ca/gui/oprofile_interface.cpp
--- CodeAnalyst-Linux-2.8.29.orig/src/ca/gui/oprofile_interface.cpp	2008-11-06 20:12:39.000000000 +0100
+++ CodeAnalyst-Linux-2.8.29.orig.detection/src/ca/gui/oprofile_interface.cpp	2009-04-07 08:41:54.000000000 +0200
@@ -112,11 +112,15 @@
 	DEFAULT_EVENT = "CPU_CLK_UNHALTED:30000:0:1:1";
 
 	// Detect number of available event counters
+	for (m_numEventCountersAvailable=MAX_EVENTNUM_MULTIPLEXING;m_numEventCountersAvailable>0;m_numEventCountersAvailable--) {
+	    if(QFile::exists(OP_MOUNT + "/" + QString::number(m_numEventCountersAvailable-1))) break;
+	}
+/*
 	if(QFile::exists(OP_MOUNT + "/" + QString::number(MAX_EVENTNUM_MULTIPLEXING-1)))
 		m_numEventCountersAvailable = MAX_EVENTNUM_MULTIPLEXING;
 	else
 		m_numEventCountersAvailable = 4;  
-
+*/
 }
 
 void oprofile_interface::set_op_events (op_event_property * properties)
