diff -dPNur CodeAnalyst-Linux-2.8.29.orig/src/ca/gui/ProfileCollection.cpp CodeAnalyst-Linux-2.8.29.orig.fix/src/ca/gui/ProfileCollection.cpp
--- CodeAnalyst-Linux-2.8.29.orig/src/ca/gui/ProfileCollection.cpp	2008-11-21 01:47:19.000000000 +0100
+++ CodeAnalyst-Linux-2.8.29.orig.fix/src/ca/gui/ProfileCollection.cpp	2009-04-07 08:31:15.000000000 +0200
@@ -585,9 +585,9 @@
 	if (!isCpuAMD ())
 	{
 		ConfigMap::Iterator it = m_configs.begin();
-		for( ; it != m_configs.end(); ++it )
+		for( ; it != m_configs.end(); )
 		{
-			QString remEvent = it.key();
+			QString remEvent = it.key(); ++it;
 			TRIGGER type = getProfileType (remEvent);
 			QString typeString;
 			if (type == EVENT_TRIGGER)
