--- a/intern/cycles/util/util_map.h
+++ b/intern/cycles/util/util_map.h
@@ -18,13 +18,13 @@
 #define __UTIL_MAP_H__
 
 #include <map>
-#include <boost/tr1/unordered_map.hpp>
+#include <boost/unordered_map.hpp>
 
 CCL_NAMESPACE_BEGIN
 
 using std::map;
 using std::pair;
-using std::tr1::unordered_map;
+using boost::unordered_map;
 
 CCL_NAMESPACE_END
 
--- a/intern/cycles/util/util_set.h
+++ b/intern/cycles/util/util_set.h
@@ -18,12 +18,12 @@
 #define __UTIL_SET_H__
 
 #include <set>
-#include <boost/tr1/unordered_set.hpp>
+#include <boost/unordered_set.hpp>
 
 CCL_NAMESPACE_BEGIN
 
 using std::set;
-using std::tr1::unordered_set;
+using boost::unordered_set;
 
 CCL_NAMESPACE_END
 
