--- godot-1.1-stable/drivers/png/SCsub.orig	2015-05-23 20:36:35.656505809 +0200
+++ godot-1.1-stable/drivers/png/SCsub	2015-05-23 20:37:10.662089045 +0200
@@ -2,22 +2,6 @@
 
 
 png_sources = [
-	"png/example.c",
-	"png/png.c",
-	"png/pngerror.c",
-	"png/pngget.c",
-	"png/pngmem.c",
-	"png/pngpread.c",
-	"png/pngread.c",
-	"png/pngrio.c",
-	"png/pngrtran.c",
-	"png/pngrutil.c",
-	"png/pngset.c",
-	"png/pngtrans.c",
-	"png/pngwio.c",
-	"png/pngwrite.c",
-	"png/pngwtran.c",
-	"png/pngwutil.c",
 	"png/resource_saver_png.cpp",
 	"png/image_loader_png.cpp"
 ]
--- godot-1.1-stable/drivers/png/image_loader_png.h.orig	2015-05-23 20:38:19.304232814 +0200
+++ godot-1.1-stable/drivers/png/image_loader_png.h	2015-05-23 20:38:48.510719513 +0200
@@ -30,7 +30,7 @@
 #define IMAGE_LOADER_PNG_H
 
 #include "io/image_loader.h"
-#include "drivers/png/png.h"
+#include <png.h>
 
 /**
 	@author Juan Linietsky <reduzio@gmail.com>
--- godot-1.1-stable/drivers/png/image_loader_png.cpp.orig	2015-05-23 20:39:48.296715869 +0200
+++ godot-1.1-stable/drivers/png/image_loader_png.cpp	2015-05-23 20:40:21.089262411 +0200
@@ -30,6 +30,7 @@
 
 #include "print_string.h"
 #include "os/os.h"
+#include <cstring>
 
 void ImageLoaderPNG::_read_png_data(png_structp png_ptr,png_bytep data, png_size_t p_length) {
 
--- godot-1.1-stable/drivers/png/resource_saver_png.cpp.orig	2015-05-23 20:44:08.165047677 +0200
+++ godot-1.1-stable/drivers/png/resource_saver_png.cpp	2015-05-23 20:44:30.429418875 +0200
@@ -28,7 +28,7 @@
 /*************************************************************************/
 #include "resource_saver_png.h"
 #include "scene/resources/texture.h"
-#include "drivers/png/png.h"
+#include <png.h>
 #include "os/file_access.h"
 #include "globals.h"
 #include "core/image.h"
--- godot-1.1-stable/drivers/ogg/SCsub.orig	2015-05-23 20:47:16.296031823 +0200
+++ godot-1.1-stable/drivers/ogg/SCsub	2015-05-23 20:47:48.136597835 +0200
@@ -1,9 +1,6 @@
 Import('env')
 
 ogg_sources = [
-
-	"ogg/bitwise.c",
-	"ogg/framing.c",
 ]
 
 if env['theora'] != "yes" or env['use_theoraplayer_binary'] != "yes":
--- godot-1.1-stable/drivers/vorbis/audio_stream_ogg_vorbis.h.orig	2015-05-23 20:48:35.394435793 +0200
+++ godot-1.1-stable/drivers/vorbis/audio_stream_ogg_vorbis.h	2015-05-23 20:49:05.129961801 +0200
@@ -30,7 +30,7 @@
 #define AUDIO_STREAM_OGG_VORBIS_H
 
 #include "scene/resources/audio_stream_resampled.h"
-#include "vorbis/vorbisfile.h"
+#include <vorbis/vorbisfile.h>
 #include "os/file_access.h"
 #include "io/resource_loader.h"
 #include "os/thread_safe.h"
--- godot-1.1-stable/drivers/vorbis/SCsub.orig	2015-05-23 20:49:30.715413651 +0200
+++ godot-1.1-stable/drivers/vorbis/SCsub	2015-05-23 20:49:52.522798247 +0200
@@ -6,31 +6,6 @@
 ]
 
 sources_lib = [
-	"vorbis/analysis.c",
-	#"vorbis/barkmel.c",
-	"vorbis/bitrate.c",
-	"vorbis/block.c",
-	"vorbis/codebook.c",
-	"vorbis/envelope.c",
-	"vorbis/floor0.c",
-	"vorbis/floor1.c",
-	"vorbis/info.c",
-	"vorbis/lookup.c",
-	"vorbis/lpc.c",
-	"vorbis/lsp.c",
-	"vorbis/mapping0.c",
-	"vorbis/mdct.c",
-	"vorbis/psy.c",
-	#"vorbis/psytune.c",
-	"vorbis/registry.c",
-	"vorbis/res0.c",
-	"vorbis/sharedbook.c",
-	"vorbis/smallft.c",
-	"vorbis/synthesis.c",
-	#"vorbis/tone.c",
-	"vorbis/vorbisenc.c",
-	"vorbis/vorbisfile.c",
-	"vorbis/window.c",
 ]
 
 env.drivers_sources += sources
--- godot-1.1-stable/drivers/mpc/SCsub.orig	2015-05-23 20:54:00.023131916 +0200
+++ godot-1.1-stable/drivers/mpc/SCsub	2015-05-23 20:54:20.231483440 +0200
@@ -2,14 +2,6 @@
 
 
 mpc_sources = [
-	"mpc/huffman.c",
-	"mpc/mpc_bits_reader.c",
-	"mpc/mpc_decoder.c",
-	"mpc/mpc_demux.c",
-	"mpc/mpc_reader.c",
-	"mpc/requant.c",
-	"mpc/streaminfo.c",
-	"mpc/synth_filter.c",
 ]
 
 env.drivers_sources+=mpc_sources
--- godot-1.1-stable/drivers/mpc/audio_stream_mpc.h.orig	2015-05-23 20:55:03.689238343 +0200
+++ godot-1.1-stable/drivers/mpc/audio_stream_mpc.h	2015-05-23 20:55:26.408632443 +0200
@@ -3,7 +3,7 @@
 
 #include "scene/resources/audio_stream_resampled.h"
 #include "os/file_access.h"
-#include "mpc/mpcdec.h"
+#include <mpc/mpcdec.h>
 #include "os/thread_safe.h"
 #include "io/resource_loader.h"
 //#include "../libmpcdec/decoder.h"
