--- drivers/mpc/SCsub.orig	2014-03-30 20:09:44.199302056 +0200
+++ drivers/mpc/SCsub	2014-03-30 20:11:18.326901233 +0200
@@ -2,14 +2,6 @@
 
 
 mpc_sources = [
-	"mpc/huffman.c",
-	"mpc/mpc_bits_reader.c",
-	"mpc/mpc_decoder.c",
-	"mpc/mpc_demux.c",
-	"mpc/mpc_reader.c",
-	"mpc/requant.c",
-	"mpc/streaminfo.c",
-	"mpc/synth_filter.c",
 ]
 
 env.drivers_sources+=mpc_sources
--- drivers/mpc/audio_stream_mpc.h.orig	2014-03-30 19:57:50.899054407 +0200
+++ drivers/mpc/audio_stream_mpc.h	2014-03-30 19:58:04.065282800 +0200
@@ -3,7 +3,7 @@
 
 #include "scene/resources/audio_stream_resampled.h"
 #include "os/file_access.h"
-#include "mpc/mpcdec.h"
+#include <mpc/mpcdec.h>
 #include "os/thread_safe.h"
 #include "io/resource_loader.h"
 //#include "../libmpcdec/decoder.h"
--- drivers/ogg/SCsub.orig	2014-03-30 20:13:44.372375812 +0200
+++ drivers/ogg/SCsub	2014-03-30 20:13:53.074523005 +0200
@@ -1,9 +1,6 @@
 Import('env')
 
 ogg_sources = [
-
-	"ogg/bitwise.c",
-	"ogg/framing.c",
 ]
 
 env.drivers_sources+=ogg_sources
--- drivers/vorbis/audio_stream_ogg_vorbis.h.orig	2014-03-30 20:55:31.006540762 +0200
+++ drivers/vorbis/audio_stream_ogg_vorbis.h	2014-03-30 20:55:54.100935388 +0200
@@ -30,7 +30,7 @@
 #define AUDIO_STREAM_OGG_VORBIS_H
 
 #include "scene/resources/audio_stream_resampled.h"
-#include "vorbis/vorbisfile.h"
+#include <vorbis/vorbisfile.h>
 #include "os/file_access.h"
 #include "io/resource_loader.h"
 #include "os/thread_safe.h"
--- drivers/vorbis/SCsub.orig	2014-03-30 20:31:18.265056556 +0200
+++ drivers/vorbis/SCsub	2014-03-30 20:31:33.396310657 +0200
@@ -3,31 +3,6 @@
 
 sources = [
 	"vorbis/audio_stream_ogg_vorbis.cpp",
-	"vorbis/analysis.c",
-	#"vorbis/barkmel.c",
-	"vorbis/bitrate.c",
-	"vorbis/block.c",
-	"vorbis/codebook.c",
-	"vorbis/envelope.c",
-	"vorbis/floor0.c",
-	"vorbis/floor1.c",
-	"vorbis/info.c",
-	"vorbis/lookup.c",
-	"vorbis/lpc.c",
-	"vorbis/lsp.c",
-	"vorbis/mapping0.c",
-	"vorbis/mdct.c",
-	"vorbis/psy.c",
-	#"vorbis/psytune.c",
-	"vorbis/registry.c",
-	"vorbis/res0.c",
-	"vorbis/sharedbook.c",
-	"vorbis/smallft.c",
-	"vorbis/synthesis.c",
-	#"vorbis/tone.c",
-	"vorbis/vorbisenc.c",
-	"vorbis/vorbisfile.c",
-	"vorbis/window.c",
 ]
 
 env.drivers_sources += sources
--- drivers/png/SCsub.orig	2014-03-31 17:39:28.060326691 +0200
+++ drivers/png/SCsub	2014-03-31 17:40:09.355918398 +0200
@@ -2,22 +2,6 @@
 
 
 png_sources = [
-	"png/example.c",
-	"png/png.c",
-	"png/pngerror.c",
-	"png/pngget.c",
-	"png/pngmem.c",
-	"png/pngpread.c",
-	"png/pngread.c",
-	"png/pngrio.c",
-	"png/pngrtran.c",
-	"png/pngrutil.c",
-	"png/pngset.c",
-	"png/pngtrans.c",
-	"png/pngwio.c",
-	"png/pngwrite.c",
-	"png/pngwtran.c",
-	"png/pngwutil.c",
 	"png/resource_saver_png.cpp",
 	"png/image_loader_png.cpp"
 ]
--- drivers/png/image_loader_png.h.orig	2014-03-30 19:40:29.839742774 +0200
+++ drivers/png/image_loader_png.h	2014-03-30 19:40:46.455035203 +0200
@@ -30,7 +30,7 @@
 #define IMAGE_LOADER_PNG_H
 
 #include "io/image_loader.h"
-#include "drivers/png/png.h"
+#include <png.h>
 
 /**
 	@author Juan Linietsky <reduzio@gmail.com>
--- drivers/png/resource_saver_png.cpp.orig	2014-03-31 17:46:40.706619538 +0200
+++ drivers/png/resource_saver_png.cpp	2014-03-31 17:46:57.313864653 +0200
@@ -28,7 +28,7 @@
 /*************************************************************************/
 #include "resource_saver_png.h"
 #include "scene/resources/texture.h"
-#include "drivers/png/png.h"
+#include <png.h>
 #include "os/file_access.h"
 
 static void _write_png_data(png_structp png_ptr,png_bytep data, png_size_t p_length) {
--- drivers/png/image_loader_png.cpp.orig	2014-03-31 17:53:27.944666298 +0200
+++ drivers/png/image_loader_png.cpp	2014-03-31 17:53:39.943845517 +0200
@@ -30,6 +30,7 @@
 
 #include "print_string.h"
 #include "os/os.h"
+#include <cstring>
 
 void ImageLoaderPNG::_read_png_data(png_structp png_ptr,png_bytep data, png_size_t p_length) {
 
