--- /usr/portage/tree/dev-ruby/sass/sass-3.1.17.ebuild	2012-05-11 22:04:24.000000000 +0200
+++ sass-3.1.19.ebuild	2012-06-22 10:46:54.606992685 +0200
@@ -1,8 +1,8 @@
 # Copyright 1999-2012 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
-# $Header: /var/cvsroot/gentoo-x86/dev-ruby/sass/sass-3.1.17.ebuild,v 1.1 2012/05/11 20:04:24 flameeyes Exp $
+# $Header: $
 
-EAPI=2
+EAPI=4
 
 USE_RUBY="ruby18 ruby19 ree18 jruby"
 
