--- assimp-3.1/code/Q3BSPZipArchive.h
+++ assimp-3.1/code/Q3BSPZipArchive.h
@@ -40,7 +40,7 @@
 #ifndef AI_Q3BSP_ZIPARCHIVE_H_INC
 #define AI_Q3BSP_ZIPARCHIVE_H_INC
 
-#include "../contrib/unzip/unzip.h"
+#include <minizip/unzip.h>
 #include "../include/assimp/IOStream.hpp"
 #include "../include/assimp/IOSystem.hpp"
 #include <string>
--- assimp-3.1/code/IFCLoader.cpp
+++ assimp-3.1/code/IFCLoader.cpp
@@ -49,7 +49,7 @@
 #include <boost/tuple/tuple.hpp>
 
 #ifndef ASSIMP_BUILD_NO_COMPRESSED_IFC
-#	include "../contrib/unzip/unzip.h"
+#	include <minizip/unzip.h>
 #endif
 
 #include "IFCLoader.h"
