--- GamePlay-3.0.0/gameplay/src/Base.h.orig	2014-08-04 15:46:22.141847691 +0200
+++ GamePlay-3.0.0/gameplay/src/Base.h	2014-08-04 15:47:51.411414541 +0200
@@ -120,8 +120,8 @@
 #endif
 
 // Bullet Physics
-#include <btBulletDynamicsCommon.h>
-#include <BulletCollision/CollisionDispatch/btGhostObject.h>
+#include <bullet/btBulletDynamicsCommon.h>
+#include <bullet/BulletCollision/CollisionDispatch/btGhostObject.h>
 #define BV(v) (btVector3((v).x, (v).y, (v).z))
 #define BQ(q) (btQuaternion((q).x, (q).y, (q).z, (q).w))
 
