--- mod_atom/configure.ac.orig	2010-11-26 11:44:52.793000011 +0100
+++ mod_atom/configure.ac	2010-11-26 12:02:11.078000030 +0100
@@ -1,56 +1,87 @@
-#                                               -*- Autoconf -*-
 # Process this file with autoconf to produce a configure script.
 
-AC_PREREQ(2.61)
+AC_PREREQ(2.65)
 AC_INIT([mod_atom], [0.0.1], [tim.bray@sun.com])
+ATOM_INIT
+
 AC_CONFIG_SRCDIR([src/mod_atom.c])
 AC_CONFIG_HEADER([config.h])
 
-AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+AM_INIT_AUTOMAKE([foreign])
 
 # Checks for programs.
-AC_PROG_CC
-AC_PROG_INSTALL
-AC_PROG_LIBTOOL
-AC_PROG_MAKE_SET
+#AC_PROG_CC
+#AC_PROG_INSTALL
+#AC_PROG_LIBTOOL
+#AC_PROG_MAKE_SET
 
 # Checks for libraries.
-AP_VERSION=2.2.4
-AP_CHECK_APACHE([$AP_VERSION], [
-  LIBTOOL="`$APR_CONFIG --apr-libtool`"
-  AC_SUBST([LIBTOOL])
+#AP_VERSION=2.2.4
+#AP_CHECK_APACHE([$AP_VERSION], [
+#  LIBTOOL="`$APR_CONFIG --apr-libtool`"
+#  AC_SUBST([LIBTOOL])
+#
+#  MODULE_CFLAGS="$AP_CFLAGS"
+#  AC_SUBST([MODULE_CFLAGS])
+#
+#  MODULE_LDFLAGS="`$APR_CONFIG --link-libtool` `$APU_CONFIG --link-libtool`"
+#  AC_SUBST([MODULE_LDFLAGS])
+#
+#  BIN_LDFLAGS="`$APR_CONFIG --link-libtool` `$APU_CONFIG --link-libtool` `$APR_CONFIG --ldflags --libs` `$APU_CONFIG --ldflags --libs`"
+#  AC_SUBST([BIN_LDFLAGS])
+#
+#  prefix="$AP_PREFIX"
+#], AC_MSG_ERROR([*** Apache version $AP_VERSION not found!]))
 
-  MODULE_CFLAGS="$AP_CFLAGS"
-  AC_SUBST([MODULE_CFLAGS])
 
-  MODULE_LDFLAGS="`$APR_CONFIG --link-libtool` `$APU_CONFIG --link-libtool`"
-  AC_SUBST([MODULE_LDFLAGS])
-
-  BIN_LDFLAGS="`$APR_CONFIG --link-libtool` `$APU_CONFIG --link-libtool` `$APR_CONFIG --ldflags --libs` `$APU_CONFIG --ldflags --libs`"
-  AC_SUBST([BIN_LDFLAGS])
+# Checks for Apache
+AP_VERSION=2.2.4
+AP_CHECK_APACHE([$AP_VERSION],
+        :,
+        AC_MSG_ERROR([*** Apache version $AP_VERSION not found!])
+        )
+#MODULE_CFLAGS="${APXS_CFLAGS} ${AP_INCLUDES} ${APR_INCLUDES}"
+#AC_SUBST(MODULE_CFLAGS)
+
+
+# Checks for APR
+APR_FIND_APR( , , 1, 1)
+if test "$apr_found" = "no"; then
+	AC_MSG_ERROR([APR could not be located. Please use the --with-apr option.])
+fi
+#CC="`$apr_config --cc`"
+#CFLAGS="$CFLAGS `$apr_config --cflags`"
+#CPPFLAGS="$CPPFLAGS `$apr_config --cppflags --includes`"
+#APR_LIBS="`$apr_config --link-ld --libs`"
+#AC_SUBST(APR_LIBS)
+#LDFLAGS="$LDFLAGS $APR_LIBS `$apr_config --ldflags`"
+#LIBTOOL="`$apr_config --apr-libtool`"
+#LIBTOOL_LIBS="$LIBTOOL_LIBS `$apr_config --link-libtool --libs`"
 
-  prefix="$AP_PREFIX"
-], AC_MSG_ERROR([*** Apache version $AP_VERSION not found!]))
 
 # Checks for header files.
 AC_HEADER_DIRENT
 AC_HEADER_STDC
 AC_CHECK_HEADERS([stdlib.h string.h])
 
+
+# Checks for HTML Tidy
 save_CPPFLAGS="$CPPFLAGS"
 CPPFLAGS="$CPPFLAGS -I/usr/include/tidy"
 AC_CHECK_HEADER([buffio.h], [
   MODULE_LDFLAGS="$MODULE_LDFLAGS -ltidy"
 ], [
-  CPPFLAGS="$save_CPPFLAGS"
+	CPPFLAGS="$save_CPPFLAGS"
+	AC_MSG_ERROR([*** HTML Tidy not found! Please install HTML Tidy (web site: http://tidy.sourceforge.net)])
 
-  AC_MSG_NOTICE([Fetching TidyLib...])
-  cvs -z3 -d:pserver:anonymous@tidy.cvs.sourceforge.net:/cvsroot/tidy co -P tidy
-  (cd tidy; sh build/gnuauto/setup.sh)
-  AC_CONFIG_SUBDIRS([tidy])
+#  AC_MSG_NOTICE([Fetching TidyLib...])
+#  cvs -z3 -d:pserver:anonymous@tidy.cvs.sourceforge.net:/cvsroot/tidy co -P tidy
+#  (cd tidy; sh build/gnuauto/setup.sh)
+#  AC_CONFIG_SUBDIRS([tidy])
+#
+#  CPPFLAGS="$CPPFLAGS -I\$(top_srcdir)/tidy/include"
+#  MODULE_LDFLAGS="$MODULE_LDFLAGS \$(top_srcdir)/tidy/src/*.lo"
 
-  CPPFLAGS="$CPPFLAGS -I\$(top_srcdir)/tidy/include"
-  MODULE_LDFLAGS="$MODULE_LDFLAGS \$(top_srcdir)/tidy/src/*.lo"
 ])
 
 # Checks for typedefs, structures, and compiler characteristics.
@@ -64,6 +95,29 @@
 AC_FUNC_STAT
 AC_CHECK_FUNCS([mkdir regcomp strdup strstr])
 
-AC_CONFIG_FILES([Makefile
-                 src/Makefile])
+AC_CONFIG_MACRO_DIR([m4])
+AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([src/Makefile])
 AC_OUTPUT
+
+#LSC_DEBUG_STATUS
+AC_MSG_NOTICE([config files successfully generated!
+
+Configuration summary for mod-atom
+
+ Install prefix .......... : $prefix
+ APache eXtenSion tool ... : $(eval echo $exec_prefix/sbin/apxs)
+ Apache module directory . : $AP_LIBEXECDIR
+ Configuration path ...... : $atom_conf_dir
+ Log File ................ : $atom_log_file
+
+ HTML Tidy support........ : $html_tidy
+
+ CFLAGS .................. : $CFLAGS
+ LDFLAGS ................. : $LDFLAGS
+ LIBS .................... : $LIBS
+
+
+ 'make' will now compile mod-atom and 'su -c make install' will install it.
+ Note: On non-Linux systems you might need to use 'gmake' instead of 'make'.
+])
