--- /dev/null	2010-11-26 11:38:56.268000000 +0100
+++ m4/mod-atom.m4	2010-11-26 12:17:32.389000023 +0100
@@ -0,0 +1,15 @@
+
+AC_DEFUN([ATOM_INIT], [
+  m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
+  AM_MAINTAINER_MODE
+
+  AC_PREFIX_DEFAULT("/usr")
+  AC_REQUIRE([AC_CANONICAL_HOST])
+
+  AC_PROG_CC_C99
+  AS_IF([test "x$ac_cv_prog_cc_c99" = "xno"], [
+    AC_MSG_ERROR([no C99 compiler found, $PACKAGE requires a C99 compiler.])
+  ])
+
+  AC_PROG_LIBTOOL
+])
