--- linux/Makefile
+++ linux/Makefile
@@ -1,6 +1,6 @@
 # $scrotwm: Makefile,v 1.4 2010/01/11 20:44:54 marco Exp $ 
 
-CFLAGS+= -Wall -ggdb -D_GNU_SOURCE -I.
+CFLAGS+= -D_GNU_SOURCE -I.
 CFLAGS+= -DSWM_LIB=\"$(LIBDIR)/libswmhack.so.$(LVERS)\"
 LDADD+= -lX11 -lXrandr
 
@@ -28,7 +28,7 @@
 	$(CC) $(CFLAGS) -c -fpic -DPIC $+ -o $@
 
 libswmhack.so.$(LVERS): swm_hack.so
-	$(CC) -shared -fpic -o libswmhack.so.$(LVERS) swm_hack.so $(LDADD)
+	$(CC) $(LDFLAGS),-soname,$@ -shared -fpic -o libswmhack.so.$(LVERS) swm_hack.so $(LDADD)
 
 install: all
 	install -m 755 -d $(DESTDIR)$(BINDIR)
