--- zeromq-2.1.3/configure.in.orig	2011-03-21 18:28:24.663000019 +0100
+++ zeromq-2.1.3/configure.in	2011-03-21 18:55:15.541000022 +0100
@@ -269,54 +269,21 @@
 AC_TYPE_UINT32_T
 AC_C_VOLATILE
 
-#  PGM extension
+# check for the presence of OpenPGM
 ac_zmq_pgm_ext="no"
+AC_ARG_WITH([pgm], AS_HELP_STRING([--with-pgm], [Build libzmq with the OpenPGM Protocol Library support [default=no]]))
 
-pgm_basename="libpgm-5.1.114~dfsg"
+AS_IF([test "x$with_pgm" = "xyes"], [
+	PKG_CHECK_MODULES([OpenPGM], [openpgm-5.1 >= 5.1])
 
-AC_SUBST(pgm_basename)
-
-AC_ARG_WITH([pgm], [AS_HELP_STRING([--with-pgm], 
-    [build libzmq with PGM extension [default=no]])], 
-    [with_pgm_ext=$withval], [with_pgm_ext=no])
-
-if test "x$with_pgm_ext" != "xno"; then
-
-    # This allows placing the tar.gz to foreign/openpgm
-    # and using ./configure --with-pgm=libpgm-x.y.z
-    if test "x$with_pgm_ext" != "xyes"; then
-        pgm_basename="$with_pgm_ext"
-    fi
-
-    # Unpack libpgm
-    AC_MSG_NOTICE([Unpacking ${pgm_basename}.tar.gz])
-    ac_zmq_pwd=`pwd`
-    cd foreign/openpgm
-
-    if ! (gzip -dc "${pgm_basename}.tar.gz" || echo "failed") | ${am__untar}; then
-        AC_MSG_ERROR([cannot unpack the foreign/openpgm/${pgm_basename}.tar.gz file])
-    fi
-
-    cd "${ac_zmq_pwd}"
-
-    if test -d foreign/openpgm/build-staging; then
-        rm -rf foreign/openpgm/build-staging
-    fi
-
-    mv foreign/openpgm/${pgm_basename} foreign/openpgm/build-staging
-    pgm_srcdir=foreign/openpgm/build-staging/openpgm/pgm
-
-    if ! test -d foreign/openpgm/build-staging/openpgm/pgm/config; then
-        mkdir foreign/openpgm/build-staging/openpgm/pgm/config
-    fi
-
-    AC_SUBST(pgm_srcdir)
-    AC_CONFIG_SUBDIRS([foreign/openpgm/build-staging/openpgm/pgm/])
-
-    #  Success!
-    AC_DEFINE(ZMQ_HAVE_OPENPGM, 1, [Have OpenPGM extension])
-    ac_zmq_pgm_ext="yes"
-fi
+	# adding define for PGM
+	AC_DEFINE(ZMQ_HAVE_OPENPGM, 1, [Have OpenPGM extension])
+	# adding CFLAGS and LIBS for PGM
+	LIBZMQ_EXTRA_CXXFLAGS="$OpenPGM_CFLAGS $LIBZMQ_EXTRA_CXXFLAGS"
+	echo "*** [ LIBZMQ_EXTRA_CXXFLAGS: ${LIBZMQ_EXTRA_CXXFLAGS} ] ***"
+	LIBS="$OpenPGM_LIBS $LIBS"
+	echo "*** [ LIBS: ${LIBS} ] ***"
+])
 
 # Set -Wall, -Werror and -pedantic
 AC_LANG_PUSH([C++])
@@ -341,51 +308,51 @@
 AC_CHECK_FUNCS(perror gettimeofday memset socket getifaddrs freeifaddrs)
 AC_CHECK_HEADERS([alloca.h])
 
-if test "x$with_pgm_ext" != "xno"; then
-
-    # Check additional functions for PGM build
-    AC_CHECK_FUNCS(poll epoll_create pselect getopt vasprintf getprotobyname_r2 getprotobyname_r)
-
-    if test "x$ac_cv_func_gettimeofday" = "xyes"; then
-        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_GETTIMEOFDAY $LIBZMQ_EXTRA_CFLAGS"
-    fi
-    if test "x$ac_cv_func_getprotobyname_r2" = "xyes"; then
-        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_GETPROTOBYNAME_R2 $LIBZMQ_EXTRA_CFLAGS"
-    fi    
-    if test "x$ac_cv_func_poll" = "xyes"; then
-        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_POLL $LIBZMQ_EXTRA_CFLAGS"
-    fi
-    if test "x$ac_cv_func_epoll_create" = "xyes"; then
-        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_EPOLL $LIBZMQ_EXTRA_CFLAGS"
-    fi
-    if test "x$ac_cv_func_getifaddrs" = "xyes"; then
-        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_GETIFADDRS $LIBZMQ_EXTRA_CFLAGS"
-    fi
-    if test "x$ac_cv_func_getopt" = "xyes"; then
-        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_GETOPT $LIBZMQ_EXTRA_CFLAGS"
-    fi
-    if test "x$ac_cv_func_pselect" = "xyes"; then
-        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_PSELECT $LIBZMQ_EXTRA_CFLAGS"
-    fi
-    if test "x$ac_cv_header_alloca_h" = "xyes"; then
-        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_ALLOCA_H $LIBZMQ_EXTRA_CFLAGS"
-    fi
-    if test "x$ac_cv_c_compiler_gnu" = "xno"; then
-        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_ISO_VARARGS $LIBZMQ_EXTRA_CFLAGS"
-    else
-        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_GNU_VARARGS $LIBZMQ_EXTRA_CFLAGS"
-    fi
-    if test "x$ac_cv_func_vasprintf" = "xyes"; then
-        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_VASPRINTF $LIBZMQ_EXTRA_CFLAGS"
-    fi
-    case "${host_cpu}" in
-        i*86|x86_64|amd64|*sparc*)
-            LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_TSC -DCONFIG_HAVE_RTC -DCONFIG_HAVE_HPET $LIBZMQ_EXTRA_CFLAGS"
-        ;;
-        *sparc*|s390*)
-        ;;
-    esac
-fi
+#if test "x$with_pgm_ext" != "xno"; then
+#
+#    # Check additional functions for PGM build
+#    AC_CHECK_FUNCS(poll epoll_create pselect getopt vasprintf getprotobyname_r2 getprotobyname_r)
+#
+#    if test "x$ac_cv_func_gettimeofday" = "xyes"; then
+#        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_GETTIMEOFDAY $LIBZMQ_EXTRA_CFLAGS"
+#    fi
+#    if test "x$ac_cv_func_getprotobyname_r2" = "xyes"; then
+#        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_GETPROTOBYNAME_R2 $LIBZMQ_EXTRA_CFLAGS"
+#    fi    
+#    if test "x$ac_cv_func_poll" = "xyes"; then
+#        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_POLL $LIBZMQ_EXTRA_CFLAGS"
+#    fi
+#    if test "x$ac_cv_func_epoll_create" = "xyes"; then
+#        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_EPOLL $LIBZMQ_EXTRA_CFLAGS"
+#    fi
+#    if test "x$ac_cv_func_getifaddrs" = "xyes"; then
+#        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_GETIFADDRS $LIBZMQ_EXTRA_CFLAGS"
+#    fi
+#    if test "x$ac_cv_func_getopt" = "xyes"; then
+#        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_GETOPT $LIBZMQ_EXTRA_CFLAGS"
+#    fi
+#    if test "x$ac_cv_func_pselect" = "xyes"; then
+#        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_PSELECT $LIBZMQ_EXTRA_CFLAGS"
+#    fi
+#    if test "x$ac_cv_header_alloca_h" = "xyes"; then
+#        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_ALLOCA_H $LIBZMQ_EXTRA_CFLAGS"
+#    fi
+#    if test "x$ac_cv_c_compiler_gnu" = "xno"; then
+#        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_ISO_VARARGS $LIBZMQ_EXTRA_CFLAGS"
+#    else
+#        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_GNU_VARARGS $LIBZMQ_EXTRA_CFLAGS"
+#    fi
+#    if test "x$ac_cv_func_vasprintf" = "xyes"; then
+#        LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_VASPRINTF $LIBZMQ_EXTRA_CFLAGS"
+#    fi
+#    case "${host_cpu}" in
+#        i*86|x86_64|amd64|*sparc*)
+#            LIBZMQ_EXTRA_CFLAGS="-DCONFIG_HAVE_TSC -DCONFIG_HAVE_RTC -DCONFIG_HAVE_HPET $LIBZMQ_EXTRA_CFLAGS"
+#        ;;
+#        *sparc*|s390*)
+#        ;;
+#    esac
+#fi
 
 # Subst LIBZMQ_EXTRA_CFLAGS & CXXFLAGS & LDFLAGS
 AC_SUBST(LIBZMQ_EXTRA_CFLAGS)
