--- zeromq-2.0.9/configure.in.orig	2010-09-18 14:23:24.702000029 +0200
+++ zeromq-2.0.9/configure.in	2010-09-18 16:26:42.291000048 +0200
@@ -273,9 +273,32 @@
 
 AC_SUBST(pgm_basename)
 
-AC_ARG_WITH([pgm], [AS_HELP_STRING([--with-pgm], 
-    [build libzmq with PGM extension [default=no]])], 
-    [with_pgm_ext=yes], [with_pgm_ext=no])
+#AC_ARG_WITH([pgm], [AS_HELP_STRING([--with-pgm], 
+#    [build libzmq with PGM extension [default=no]])], 
+#    [with_pgm_ext=yes], [with_pgm_ext=no])
+
+AC_ARG_WITH([pgm],
+	AS_HELP_STRING(
+		[--without-pgm],
+		[Ignore presence of OpenPGM and disable it]
+	)
+)
+
+AS_IF([test "x$with_pgm" != "xno"],
+	[
+		AC_MSG_CHECKING([for OpenPGM ** ])
+		PKG_CHECK_MODULES(PGM, [ libpgm >= 5.0.78 ], [ have_pgm=yes ], [ have_pgm=no ])
+	],
+	[ have_pgm=no ]
+)
+
+AS_IF([test "x$have_pgm" = "xyes"],
+	[FUFFA OpenPGM **],
+	[AS_IF([test "x$with_pgm" = "xyes"],
+		[AC_MSG_ERROR([OpenPGM requested but not found])
+	])
+])
+
 
 if test "x$with_pgm_ext" != "xno"; then
 
