--- rdoc.rake.orig	2013-02-22 15:26:46.911131298 +0100
+++ rdoc.rake	2013-02-22 15:28:19.845126578 +0100
@@ -1,18 +1,18 @@
-require 'rake/rdoctask'
+require 'rdoc/task'
 
 CLEAN.include %w(doc/rdoc)
 
-Rake::RDocTask.new do |rdoc|
+RDoc::Task.new do |rdoc|
   rdoc.rdoc_dir = 'doc/rdoc'
   rdoc.options += ['--quiet', '--title', Thin::NAME,
              	     "--opname", "index.html",
             	     "--line-numbers",
-            	     "--main", "README",
+            	     "--main", "README.md",
             	     "--inline-source"]
   rdoc.template = "site/rdoc.rb"
-  rdoc.main = "README"
+  rdoc.main = "README.md"
   rdoc.title = Thin::NAME
-  rdoc.rdoc_files.add %w(README) +
+  rdoc.rdoc_files.add %w(README.md) +
                       FileList['lib/**/*.rb'] +
                       FileList['bin/*']
 end
