diff -Naur panda3d-1.8.0.old/makepanda/makepanda.py panda3d-1.8.0/makepanda/makepanda.py
--- panda3d-1.8.0.old/makepanda/makepanda.py	2012-02-13 02:17:47.000000000 -0500
+++ panda3d-1.8.0/makepanda/makepanda.py	2012-06-05 15:59:57.000000000 -0400
@@ -606,12 +606,12 @@
 
     #         Name         pkg-config   libs, include(dir)s
     if (not RUNTIME):
-        SmartPkgEnable("EIGEN",     "",          (), ("Eigen/Dense",), target_pkg = 'ALWAYS')
+        SmartPkgEnable("EIGEN",     "eigen3",    (), ("Eigen/Dense",), target_pkg = 'ALWAYS')
         SmartPkgEnable("ARTOOLKIT", "",          ("AR"), "AR/ar.h")
         SmartPkgEnable("FCOLLADA",  "",          ChooseLib(*fcollada_libs), ("FCollada", "FCollada.h"))
         SmartPkgEnable("FFMPEG",    ffmpeg_libs, ffmpeg_libs, ffmpeg_libs)
         SmartPkgEnable("SWSCALE",   "libswscale", "libswscale", ("libswscale", "libswscale/swscale.h"), target_pkg = "FFMPEG")
-        SmartPkgEnable("FFTW",      "",          ("fftw", "rfftw"), ("fftw.h", "rfftw.h"))
+        SmartPkgEnable("FFTW",      "fftw2",     ("fftw", "rfftw"), ("fftw.h", "rfftw.h"))
         SmartPkgEnable("FMODEX",    "",          ("fmodex"), ("fmodex", "fmodex/fmod.h"))
         SmartPkgEnable("FREETYPE",  "freetype2", ("freetype"), ("freetype2", "freetype2/freetype/freetype.h"))
         SmartPkgEnable("GL",        "gl",        ("GL"), ("GL/gl.h"), framework = "OpenGL")
diff -Naur panda3d-1.8.0.old/makepanda/makepandacore.py panda3d-1.8.0/makepanda/makepandacore.py
--- panda3d-1.8.0.old/makepanda/makepandacore.py	2012-01-28 08:12:36.000000000 -0500
+++ panda3d-1.8.0/makepanda/makepandacore.py	2012-06-05 15:18:55.000000000 -0400
@@ -1005,7 +1005,10 @@
     if (sys.platform == "win32" or not LocateBinary(tool)):
         return []
     if (tool == "pkg-config"):
-        handle = os.popen(LocateBinary("pkg-config") + " --silence-errors --libs-only-l " + pkgname)
+        if (pkgname == "opencv"):
+            handle = os.popen(LocateBinary("pkg-config") + " --silence-errors --libs " + pkgname)
+        else:
+            handle = os.popen(LocateBinary("pkg-config") + " --silence-errors --libs-only-l " + pkgname)
     elif (tool == "fltk-config"):
         handle = os.popen(LocateBinary("fltk-config") + " --ldstaticflags")
     else:
