--- ../../../../distfiles/cvs-src/panda3d/panda/src/vision/webcamVideoCursorV4L.cxx	2010-07-28 10:36:11.000000000 +0200
+++ panda3d-patched/panda/src/vision/webcamVideoCursorV4L.cxx	2010-08-07 21:18:53.443409116 +0200
@@ -25,8 +25,9 @@
 #ifdef SUPPORT_WEBCAM_VIDEO_JPEG
 extern "C" {
   #include <jpeglib.h>
-  #include <jpegint.h>
+  //#include <jpegint.h>
   #include <jerror.h>
+  #define DSTATE_READY    202
 }
 
 #include <setjmp.h>
