--- recoll-1.19.14p1/lib/mkMake.in
+++ recoll-1.19.14p1/lib/mkMake.in
@@ -142,7 +142,8 @@
 # no effort whatsoever to maintain any kind of ABI compat in this lib. Only
 # programs or modules from the same Recoll release can be compatible.
 \$(LIBRECOLL): \$(DEPS) \$(OBJS)
-	\$(CXX) -shared -Wl,--no-undefined -Wl,--warn-unresolved-symbols \
+	\$(CXX) \$(ALL_CXXFLAGS) \$(LDFLAGS) \
+	-shared -Wl,--no-undefined -Wl,--warn-unresolved-symbols \
           -Wl,-soname=\$(LIBRECOLL) -o \$(LIBRECOLL) \$(OBJS) \
            \$(LIBXAPIAN) \$(LIBICONV) \$(X_LIBX11) \$(LIBSYS)
 	rm -f librecoll.so
@@ -153,12 +154,12 @@
 for c in $SRC_CPP;do
     o=`basename $c .cpp`.o
     echo "$o : $c $defs" >> $mk
-    echo "	\$(CXX) \$(ALL_CXXFLAGS) -c $c" >> $mk
+    echo "	\$(CXX) \$(ALL_CXXFLAGS) \$(LDFLAGS) -c $c" >> $mk
 done
 for c in $SRC_CC;do
     o=`basename $c .cc`.o
     echo "$o : $c $defs" >> $mk
-    echo "	\$(CXX) \$(ALL_CXXFLAGS) -c $c" >> $mk
+    echo "	\$(CXX) \$(ALL_CXXFLAGS) \$(LDFLAGS) -c $c" >> $mk
 done
 
 cat >> $mk <<EOF
@@ -174,7 +175,7 @@
     d=`basename $c .cpp`.dep
     s=`basename $c .cpp`.dep.stamp
     echo "$s : $c $defs" >> $mk
-    echo "	\$(CXX) -M \$(ALL_CXXFLAGS) $c > $d" >> $mk
+    echo "	\$(CXX) -M \$(ALL_CXXFLAGS) \$(LDFLAGS) $c > $d" >> $mk
     echo "	touch $s" >> $mk
 done
 for c in $SRC_CPP;do
@@ -185,7 +186,7 @@
     d=`basename $c .cc`.dep
     s=`basename $c .cc`.dep.stamp
     echo "$s : $c $defs" >> $mk
-    echo "	\$(CXX) -M \$(ALL_CXXFLAGS) $c > $d" >> $mk
+    echo "	\$(CXX) -M \$(ALL_CXXFLAGS) \$(LDFLAGS) $c > $d" >> $mk
     echo "	touch $s" >> $mk
 done
 for c in $SRC_CC;do
