--- SFML-2.1/CMakeLists.txt
+++ SFML-2.1/CMakeLists.txt
@@ -186,9 +186,9 @@
             COMPONENT devel)
 endif()
 
-install(FILES cmake/Modules/FindSFML.cmake DESTINATION ${INSTALL_MISC_DIR}/cmake/Modules)
-install(FILES license.txt DESTINATION ${INSTALL_MISC_DIR})
-install(FILES readme.txt DESTINATION ${INSTALL_MISC_DIR})
+install(FILES cmake/Modules/FindSFML.cmake DESTINATION /usr/share/cmake/Modules)
+#install(FILES license.txt DESTINATION ${INSTALL_MISC_DIR})
+#install(FILES readme.txt DESTINATION ${INSTALL_MISC_DIR})
 
 if(WINDOWS)
     if(ARCH_32BITS)
