--- netsurf-3.1/Makefile.defaults	2014-05-31 12:05:30.498531102 +0200
+++ netsurf-3.1/Makefile.defaults	2014-05-31 12:06:03.539079625 +0200
@@ -87,7 +87,7 @@
 NETSURF_USE_LIBICONV_PLUG := YES
 
 # Initial CFLAGS. Optimisation level etc. tend to be target specific.
-CFLAGS :=
+CFLAGS := $(CFLAGS)
 
 # Default installation/execution prefix
 PREFIX ?= /usr/local
--- netsurf-3.1/framebuffer/Makefile.defaults	2014-05-31 12:08:30.278515450 +0200
+++ netsurf-3.1/framebuffer/Makefile.defaults	2014-05-31 12:08:53.133894825 +0200
@@ -3,7 +3,6 @@
 # ----------------------------------------------------------------------------
 
 # Optimisation levels
-CFLAGS += -O2
 
 # Framebuffer default surface provider.
 # Valid values are: x, sdl, linux, vnc, able,
