--- dmd/src/dmd/posix.mak
+++ dmd/src/dmd/posix.mak
@@ -29,7 +29,7 @@
 TK=tk
 ROOT=root
 
-MODEL=32
+MODEL?=32
 
 ifeq (OSX,$(TARGET))
     ## See: http://developer.apple.com/documentation/developertools/conceptual/cross_development/Using/chapter_3_section_2.html#//apple_ref/doc/uid/20002000-1114311-BABGCAAB
@@ -42,12 +42,12 @@
     #-syslibroot is only passed to libtool, not ld.
     #if gcc sees -isysroot it should pass -syslibroot to the linker when needed
     #LDFLAGS=-lstdc++ -isysroot ${SDK} -Wl,-syslibroot,${SDK} -framework CoreServices
-    LDFLAGS=-lstdc++ -isysroot ${SDK} -Wl -framework CoreServices
+    LDFLAGS +=-lstdc++ -isysroot ${SDK} -Wl -framework CoreServices
 else
-    LDFLAGS=-lm -lstdc++ -lpthread
+    LDFLAGS +=-lm -lstdc++ -lpthread
 endif
 
-CC=g++ -m$(MODEL) $(TARGET_CFLAGS)
+CC=$(CXX) -m$(MODEL) $(TARGET_CFLAGS)
 
 #OPT=-g -g3
 #OPT=-O2
@@ -59,8 +59,8 @@
 #GFLAGS = $(WARNINGS) -D__near= -D__pascal= -fno-exceptions -g -DDEBUG=1 -DUNITTEST $(COV)
 GFLAGS = $(WARNINGS) -D__near= -D__pascal= -fno-exceptions -O2
 
-CFLAGS = $(GFLAGS) -I$(ROOT) -DMARS=1 -DTARGET_$(TARGET)=1 -D_DH
-MFLAGS = $(GFLAGS) -I$C -I$(TK) -DMARS=1 -DTARGET_$(TARGET)=1 -D_DH
+MFLAGS = $(CFLAGS) $(GFLAGS) -I$C -I$(TK) -DMARS=1 -DTARGET_$(TARGET)=1 -D_DH
+CFLAGS += $(GFLAGS) -I$(ROOT) -DMARS=1 -DTARGET_$(TARGET)=1 -D_DH
 
 CH= $C/cc.h $C/global.h $C/oper.h $C/code.h $C/type.h \
 	$C/dt.h $C/cgcv.h $C/el.h $C/iasm.h
@@ -141,7 +141,7 @@
 all: dmd
 
 dmd: $(DMD_OBJS)
-	$(ENVP) gcc -o dmd -m$(MODEL) $(COV) $(DMD_OBJS) $(LDFLAGS)
+	$(ENVP) $(CC) -o dmd -m$(MODEL) $(COV) $(DMD_OBJS) $(LDFLAGS)
 
 clean:
 	rm -f $(DMD_OBJS) dmd optab.o id.o impcnvgen idgen id.c id.h \
@@ -511,7 +511,7 @@
 	$(CC) -c $(GFLAGS) -I$(ROOT) $<
 
 strtold.o: $C/strtold.c
-	gcc -m$(MODEL) -c $<
+	$(CC) -m$(MODEL) -c $<
 
 struct.o: struct.c
 	$(CC) -c $(CFLAGS) $<
--- dmd/src/phobos/linux.mak
+++ dmd/src/phobos/linux.mak
@@ -12,17 +12,17 @@
 LIB=lib$(MODEL)/libphobos.a
 
 MAKEFILE=linux.mak
-MODEL=32
+MODEL ?= 32
 
-CFLAGS=-O -m$(MODEL)
+CFLAGS +=-O -m$(MODEL)
 #CFLAGS=-g -m$(MODEL)
 
 DFLAGS=-O -release -w -m$(MODEL)
 #DFLAGS=-unittest -w -m$(MODEL)
 
-CC=gcc
+CC ?= $(CC)
 #DMD=/dmd/bin/dmd
-DMD=dmd
+DMD ?= dmd
 
 .SUFFIXES: .c .o .cpp .d .asm
 
@@ -30,7 +30,7 @@
 	$(CC) -c $(CFLAGS) $*.c
 
 .cpp.o:
-	g++ -c $(CFLAGS) $*.cpp
+	${CXX} -c $(CFLAGS) $*.cpp
 
 .d.o:
 	$(DMD) -c $(DFLAGS) $*.d
